lib/improver/wind_downscaling.py

```diff
@@ -652,8 +661,14 @@ def __init__(self, a_over_s_cube, sigma_cube, pporo_cube,
         self.sigma = next(sigma_cube.slices([y_name, x_name]))
         try:
             self.z_0 = next(z0_cube.slices([y_name, x_name]))
+            self.z_0.data = np.float32(self.z_0.data)
         except AttributeError:
             self.z_0 = z0_cube
+
+        if height_levels_cube is not None:
+            height_levels_cube.data = height_levels_cube.data.astype(
```

Comment:
This could be written as, which might be slightly better:
```python
height_levels_cube.data = (
    height_levels_cube.data.astype(np.float32))
```